src/ui/events.js

```diff
@@ -251,35 +307,47 @@ export type MapBoxZoomEvent = {
 };
 
 /**
- * A `MapDataEvent` object is emitted with the {@link Map.event:data}
- * and {@link Map.event:dataloading} events. Possible values for
- * `dataType`s are:
- *
- * - `'source'`: The non-tile data associated with any source
- * - `'style'`: The [style](https://www.mapbox.com/mapbox-gl-style-spec/) used by the map
+ * `MapDataEvent` is a class used by other classes to generate 
+ * mouse events of specific types such as 'sourcedata' or 'dataloading'.
+ * For a full list of available events, see [`Map` events](/mapbox-gl-js/api/map/#map-events).
  *
  * @typedef {Object} MapDataEvent
- * @property {string} type The event type.
- * @property {string} dataType The type of data that has changed. One of `'source'`, `'style'`.
+ * @property {string} type The type of originating event. For a full list of available events, see [`Map` events](/mapbox-gl-js/api/map/#map-events).
+ * @property {string} dataType The type of data that has changed. One of `'source'` or `'style'`, where `'source'` refers to the non-tile data associated with any source, and `'style'` refers to the [style](https://docs.mapbox.com/help/glossary/style/) used by the map.
```

Comment:
I think I know what you're going for, but "non-tile data" may be a bit confusing. Source data can be tiles (vector or raster tiles and GeoJSON sources are tiled in the backend). I think the distinction is if you change the entire style, you'll get the `style` type and if you just update data within a style, you'll get `source`.

Reply:
Great that we can make this more clear – thank you @ryanhamley!

I've changed this in 6a7209d – your edits are welcome if you see room for more improvement 👍🏼